components/apimgt/org.wso2.carbon.apimgt.event.output.adapter.http.extended.oauth/src/main/java/org/wso2/carbon/apimgt/event/output/adapter/http/extended/oauth/ExtendedHTTPEventAdapter.java

```diff
@@ -0,0 +1,367 @@
+            } catch (Exception e) {
+                log.warn("Header property '" + header + "' is not defined in the correct format.", e);
+            }
+        }
+        return result;
+
+    }
+
+    /**
+     * This class represents a job to send an HTTP request to a target URL.
+     */
+    class HTTPSender implements Runnable {
+
+        private String url;
+
+        private String payload;
+
+        private String accessToken;
+
+        private String userName;
+
+        private String password;
+
+        private Map<String, String> headers;
+
+        private HttpClient httpClient;
+
+        public HTTPSender(String url, String payload, String accessToken, Map<String, String> headers,
+                          HttpClient httpClient) {
+            this.url = url;
+            this.payload = payload;
+            this.accessToken = accessToken;
+            this.headers = headers;
+            this.httpClient = httpClient;
+        }
+
+        /**
+         * If user name and password is given, basic auth is used. If not OAuth2 is used.
+         */
+        public HTTPSender(String url, String payload, String userName, String password, Map<String, String> headers,
+                          HttpClient httpClient) {
+            this.url = url;
+            this.payload = payload;
+            this.userName = userName;
+            this.password = password;
+            this.headers = headers;
+            this.httpClient = httpClient;
+        }
+
+        public String getUrl() {
+            return url;
+        }
+
+        public String getPayload() {
+            return payload;
+        }
+
+        public String getAccessToken() {
+            return accessToken;
+        }
+
+        public Map<String, String> getHeaders() {
+            return headers;
+        }
+
+        public HttpClient getHttpClient() {
+            return httpClient;
+        }
+
+        public String getUserName() {
+            return userName;
+        }
+
+        public String getPassword() {
+            return password;
+        }
+
+        public void run() {
+
+            EntityEnclosingMethod method = null;
+
+            try {
+                if (clientMethod.equalsIgnoreCase(ExtendedHTTPEventAdapterConstants.CONSTANT_HTTP_PUT)) {
+                    method = new PutMethod(this.getUrl());
+                } else {
+                    method = new PostMethod(this.getUrl());
+                }
+
+                if (hostConfiguration == null) {
+                    URL hostUrl = new URL(this.getUrl());
+                    hostConfiguration = new HostConfiguration();
+                    hostConfiguration.setHost(hostUrl.getHost(), hostUrl.getPort(), hostUrl.getProtocol());
+                }
+
+                method.setRequestEntity(new StringRequestEntity(this.getPayload(), contentType, "UTF-8"));
+
+                if (this.getAccessToken() != null && this.getAccessToken().trim().length() > 0) {
```

Comment:
We can use isBlank() or isEmpty() frunction in https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/StringUtils.html

Reply:
Fixed